qiskit_nature/second_q/mappers/bosonic_logarithmic_mapper.py

```diff
@@ -0,0 +1,266 @@
+                        elif f"{final_state[j]}{init_state[j]}" == "10":
+                            single_mapped_term = single_mapped_term.compose(
+                                self._get_single_qubit_pauli_matrix(
+                                    mode_index_in_register, qubit_register_length, i, "S-"
+                                )
+                            )
+                        else:
+                            raise ValueError(f"Invalid state {final_state[i]}{init_state[i]}.")
+                    pauli_expansion.append(prefactor * single_mapped_term)
+                # Add the Pauli expansion for a single n_k to map of the bosonic operator
+                bos_op_to_pauli_op = reduce(operator.add, pauli_expansion).compose(
+                    bos_op_to_pauli_op
+                )
+            # Add the map of the single boson op (e.g. +_0) to the map of the full bosonic operator
+            pauli_op.append(coeff * reduce(operator.add, bos_op_to_pauli_op.simplify()))
+        # return the lookup table for the transformed XYZI operators
+        return reduce(operator.add, pauli_op)
+
+    @lru_cache(maxsize=32)
+    def _get_binary_state(self, n: int) -> str:
+        """This method converts an integer to its binary representation."""
+        return bin(n).split("b")[1].rjust(self.number_of_qubits_per_mode, "0")
+
+    @lru_cache(maxsize=32)
+    def _get_single_qubit_pauli_matrix(
+        self, register_index: int, register_length: int, mode_qubit_index: int, pauli_op: str
+    ) -> SparsePauliOp:
+        """This method builds the Qiskit Pauli operators of one of the operators:
+        I_+ = I + Z, I_- = I - Z, S_+ = X + iY and S_- = X - iY.
+
+        Args:
+            register_index: the index of the qubit register where the mapped operator should be placed.
+            register_length: the length of the qubit register.
+            mode_qubit_index: the index of the qubit within the mode on which the `pauli_op` is acting.
+            pauli_op: the operator to be mapped. Possible values are 'I+', 'I-', 'S+' and 'S-'.
+
+        Returns:
+            A SparsePauliOp representing the Pauli operator.
+        """
+        # Define recurrent variables
+        prefix_zeros = [0] * register_index + [0] * mode_qubit_index
+        suffix_zeros = [0] * (register_length - self.number_of_qubits_per_mode - register_index) + [
+            0
+        ] * (self.number_of_qubits_per_mode - mode_qubit_index - 1)
+        if pauli_op in ("I+", "I-"):
+            identity = Pauli(
+                (
+                    [0] * register_length,
+                    [0] * register_length,
+                )
+            )
+            sigma_z = Pauli(
+                (
+                    prefix_zeros + [1] + suffix_zeros,
+                    [0] * register_length,
+                )
+            )
+            if pauli_op == "I+":
+                return 0.5 * (SparsePauliOp(identity) + SparsePauliOp(sigma_z))
```

Comment:
A much simpler way for this is using `SparsePauliOp.from_sparse_list`. For example:
```python
return SparsePauliOp.from_sparse_list([("", [], 0.5), ("Z", [1], 0.5)], num_qubits=4)
```
The above gives you `I+` acting on qubit index 1 out of 4.

This should allow you to simplify the implementation of this entire method.

Reply:
Indeed it's much cleaner now!